lib/ini.js

```diff
@@ -104,16 +123,30 @@ function splitSections (str, separator) {
 }
 
 const decode = (str, opt = {}) => {
+  // The `typeof` check is required because accessing the `process` directly fails on browsers.
+  /* istanbul ignore next */
+  opt.platform = opt.platform || (typeof process !== 'undefined' && process.platform)
+  /* istanbul ignore next */
+  const eol = opt.platform === 'win32' ? '\r\n' : '\n'
   opt.bracketedArray = opt.bracketedArray !== false
   const out = Object.create(null)
   let p = out
   let section = null
+  const preserveComments = opt.preserveComments || false
+  const commentsKey = opt.commentsKey || 'comments'
+  const commentsDictionary = {}
+  let lineCommentArray = []
+  const commentsRegEx = /^[#;]+[-/\\|~`'":,.;!<>?{}[\]@#$%^&*+=()\w ]+$/
```

Comment:
This appears to be susceptible to a polynomial ReDOS https://codeql.github.com/codeql-query-help/javascript/js-polynomial-redos/